packages/config/src/utils/extensions/auto-install-extensions.ts

```diff
@@ -0,0 +1,138 @@
+import { promises as fs } from 'fs'
+
+import { NetlifyAPI } from 'netlify'
+
+import { JIGSAW_URL } from './constants.js'
+import {
+  getExtension,
+  getInstalledExtensionsForSite,
+  installExtension,
+  InstallExtensionResult,
+  getAccount,
+} from './utils.js'
+
+export async function handleAutoInstallExtensions(opts: $TSFixMe, api?: NetlifyAPI) {
+  if (!opts.accountId || !opts.token || !api?.accessToken) {
+    // skip installing extensions if not logged in
+    return
+  }
+
+  const installableExtensions = await getExtensionsToInstall(opts, api)
+  if (installableExtensions.length === 0) {
+    return
+  }
+
+  const accountId = opts.accountId
+  const account = await getAccount(api, { accountId })
+
+  const results: InstallExtensionResult[] = await Promise.all(
+    installableExtensions.map((installableExt) => {
+      if (!installableExt.hostSiteUrl) {
```

Comment:
We don't have hostSiteUrl in that endpoint right now, should we first add it? 

https://github.com/netlify/jigsaw/pull/561/files

Reply:
Yeah I'll add it there in jigsaw so we don't need to make an additional request.

Reply:
😄Added hostSiteUrl and cache control headers in this PR: https://github.com/netlify/jigsaw/pull/566